src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx

```diff
@@ -128,15 +128,22 @@ const GradingSubmissionTable: React.FC<GradingSubmissionTableProps> = ({
     ...tableFilters.columnFilters
   ]);
 
-  // Polish: debounce this search, or have a onClick event listener search instead.
-  // Polish: if search value does not change content of submissions, do not reset page.
-  // not as easy as i thought it was with setTimeout.
-  const [searchValue, setSearchValue] = useState('');
+  const [visibleSearchValue, setVisibleSearchValue] = useState('');
+  const [debouncedSearchValue, setDebouncedSearchValue] = useState('');
+
+  // timeout delay could be a placed within a config.
+  useEffect(() => {
+    const changeSearch = setTimeout(() => {
+      setDebouncedSearchValue(visibleSearchValue);
+    }, 300);
+
+    return () => clearTimeout(changeSearch);
+  }, [visibleSearchValue]);
```

Comment:
While this debounce implementation works, let's not do this manually. Lodash already provides a `debounce` helper: <https://lodash.com/docs/#debounce>